tor/net.go

```diff
@@ -20,10 +23,15 @@ const (
 // interface.
 type DialFunc func(net, addr string, timeout time.Duration) (net.Conn, error)
 
+type Dialer interface {
+	Dial(network, address string) (net.Conn, error)
+}
+
 // Net is an interface housing a Dial function and several DNS functions that
 // allows us to abstract the implementations of these functions over different
 // networks, e.g. clearnet, Tor net, etc.
 type Net interface {
+	createDialer(timeout time.Duration) (Dialer, error)
```

Comment:
Adding a non-exported method to an interface means that nobody outside of this package can implement it, which is not desired. Can't this be solved in a different way?